packages/@lwc/babel-plugin-component/src/decorators/index.js

```diff
@@ -216,6 +221,19 @@ function decorators({ types: t }) {
                 return;
             }
 
+            if (
+                node.superClass === null &&
+                isAPIFeatureEnabled(
+                    AVOID_DECORATORS_FOR_NON_LIGHTNING_ELEMENT_CLASSES,
+                    getAPIVersionFromNumber(state.opts.apiVersion)
+                )
```

Comment:
TODO: should probably avoid calling `getAPIVersionFromNumber` over and over, but I wasn't quite sure where to put it.